cloudformation/cloudwatchDashboard.yaml

```diff
@@ -0,0 +1,9 @@
+Resources:
+  CloudWatchDashboard:
+    Type: AWS::CloudWatch::Dashboard
+    Properties:
+      DashboardBody: !Sub
+        - '{ "widgets": [ { "height": 10, "width": 24, "y": 0, "x": 0, "type": "metric", "properties": { "metrics": [ [ "AWS/ApiGateway", "Latency", "ApiName", "${ApiName}", { "stat": "p50", "label": "50th" } ], [ "...", { "stat": "p90", "label": "90th" } ], [ "...", { "label": "99th", "color": "#d62728" } ] ], "view": "timeSeries", "stacked": false, "region": "${AWS::Region}", "stat": "p99", "period": 900, "title": "Resource API Latency", "annotations": { "horizontal": [ { "color": "#2ca02c", "value": 500, "fill": "below" } ] } } }, { "height": 10, "width": 12, "y": 10, "x": 0, "type": "metric", "properties": { "metrics": [ [ "AWS/ApiGateway", "Count", "ApiName", "${ApiName}", { "id": "count" } ] ], "view": "timeSeries", "stacked": false, "region": "${AWS::Region}", "stat": "Sum", "period": 900, "title": "Resource API Count", "yAxis": { "left": { "label": "", "showUnits": true } } } }, { "height": 10, "width": 12, "y": 10, "x": 12, "type": "metric", "properties": { "metrics": [ [ { "expression": "100*(client_error/count)", "label": "4XXError", "id": "e1", "region": "${AWS::Region}" } ], [ { "expression": "100*(server_error/count)", "label": "5XXError", "id": "e2", "region": "${AWS::Region}" } ], [ "AWS/ApiGateway", "Count", "ApiName", "${ApiName}", { "id": "count", "visible": false } ], [ ".", "4XXError", ".", ".", { "id": "client_error", "color": "#ff7f0e", "visible": false } ], [ ".", "5XXError", ".", ".", { "id": "server_error", "color": "#d62728", "visible": false } ] ], "view": "timeSeries", "stacked": false, "region": "${AWS::Region}", "stat": "Sum", "period": 900, "title": "Resource API Error Rate", "yAxis": { "left": { "label": "Percent", "showUnits": false, "min": 0, "max": 100 } } } }, { "type": "metric", "x": 0, "y": 20, "width": 12, "height": 10, "properties": { "metrics": [ [ "AWS/DynamoDB", "SuccessfulRequestLatency", "TableName", "${ResourceDynamoDBTableV2}", "Operation", "TransactWriteItems" ], [ "...", "PutItem" ], [ "...", "UpdateItem" ], [ "...", "Query", { "visible": false } ] ], "view": "timeSeries", "stacked": false, "region": "${AWS::Region}", "title": "Resource DB Latency (Avg)", "stat": "Average", "period": 900 } }, { "type": "metric", "x": 12, "y": 20, "width": 12, "height": 10, "properties": { "metrics": [ [ "AWS/ES", "SearchLatency", "DomainName", "${ElasticSearchDomain}", "ClientId", "${AWS::AccountId}" ] ], "view": "timeSeries", "stacked": false, "region": "${AWS::Region}", "stat": "Average", "period": 900, "title": "Resource ES Latency (Avg)" } } ] }'
+        - ApiName: !Join ['-', [!Ref Stage, '${self:service}']]
```

Comment:
Based on the latest discussion in the task, we should switch to monitoring the private API Gateway instance instead of the public one.

Reply:
done.

Reply:
For the time-being, I am referencing original API gateway not the private API gateway as private API gateway feature has been disabled.